src/main/groovy/life/qbic/oncostore/model/Case.groovy

```diff
@@ -1,43 +1,55 @@
 package life.qbic.oncostore.model
 
 import com.fasterxml.jackson.annotation.JsonProperty
+import groovy.transform.EqualsAndHashCode
 import io.swagger.v3.oas.annotations.media.Schema
 
-@Schema(name="Case", description="A case")
+/**
+ * A case (patient)
+ *
+ * @since: 1.1.0
+ */
+@EqualsAndHashCode
+@Schema(name = "Case", description = "A case")
 class Case {
 
-        String identifier
-        String projectId
-
-        Case(String identifier, String projectId) {
-            this.identifier = identifier
-            this.projectId = projectId
-        }
-
-        Case(String identifier) {
-            this.identifier = identifier
-        }
-
-        Case() {
-        }
-
-        @Schema(description="The case identifier")
-        @JsonProperty("id")
-        String getIdentifier() {
-            return identifier
-        }
-
-        @Schema(description="The associated project identifier")
-        @JsonProperty("projectId")
-        String getProjectId() {
-            return projectId
-        }
-
-        void setIdentifier(String identifier) {
-            this.identifier = identifier
-        }
-
-        void setProjectId(String projectId) {
-            this.projectId = projectId
-        }
+    /**
+     * The identifier of a case
+     */
+    String identifier
+    /**
+     * The project identifier associated with a case
+     */
+    String projectId
+
+    Case(String identifier, String projectId) {
+        this.identifier = identifier
+        this.projectId = projectId
+    }
+
+    Case(String identifier) {
+        this.identifier = identifier
+    }
+
+    Case() {}
```

Comment:
When providing overloaded constructors, it would be nice to set the other properties to a default or empty, otherwise you will provoke NullPointers when accessing the properties.

Reply:
Thanks for the hint!